Source/DafnyCore/AST/Types/UserDefinedType.cs

```diff
@@ -400,6 +400,12 @@ public override bool SupportsEquality {
           i++;
         }
         return true;
+      } else if (ResolvedClass is NewtypeDecl newtypeDecl) {
+        if (newtypeDecl.IsRevealedInScope(Type.GetScope())) {
+          return newtypeDecl.RhsWithArgument(TypeArgs).SupportsEquality;
```

Comment:
Because of my change, we could also do the same treatment as for the datatypes, if you preferred. I'm fine with both approaches and to switch if you prefer.